lib/p5.play.js

```diff
@@ -1241,6 +1241,36 @@ function Sprite(pInst, _x, _y, _w, _h) {
   */
   this.animation = undefined;
 
+  /**
+   * Swept collider oriented along the current velocity vector, extending to
+   * cover the old and new positions of the sprite.
+   *
+   * The corners of the swept collider will extend beyond the actual swept
+   * shape, but it should be sufficient for broad-phase detection of collision
+   * candidates.
+   *
+   * Note that this collider will have no dimensions if the source sprite has no
+   * velocity.
+   */
+  this._sweptCollider = new p5.OrientedBoundingBoxCollider();
+
+  /**
+   * If the sprite is moving, use the swept collider. Otherwise use the actual
+   * collider.
+   */
+  this._getBroadPhaseCollider = function() {
+    return (this.velocity.magSq() > 0) ? this._sweptCollider : this.collider;
+  };
```

Comment:
As for the "right" way to handle all of this to avoid as many edge cases as possible, the gold standard that I know of is "conservative advancement" - rather than handling collisions pairwise, we basically look at the whole world, and for any fast-moving object, expand its AABB (or OBB, depending on your implementation) sort of like in this PR.

Then, instead of merely reporting collisions between objects, you actually solve for time-of-impact (TOI) - you can see Box2d's implementation here (https://github.com/erincatto/Box2D/blob/master/Box2D/Box2D/Collision/b2TimeOfImpact.cpp#L254), it's pretty nasty work, esp. since underneath the hood it relies on the GJK algorithm (https://en.wikipedia.org/wiki/Gilbert%E2%80%93Johnson%E2%80%93Keerthi_distance_algorithm, implemented at https://github.com/erincatto/Box2D/blob/master/Box2D/Box2D/Collision/b2Distance.cpp) to figure out distances between shapes.

After that, you sort all your collisions by TOI, and step the whole world (or at least a relevant "island" of object in it) until the first collision happens, then you resolve that collision, then rinse and repeat until there are no more TOI collisions left in your timestep.

In our case, though, we have simplifications, in that there's no angular velocity and we have a limited set of primitives, so we can *probably* figure out a simpler way to get exact or at least conservative TOIs for our objects, and at the very least skip a lot of the multisampling steps (if we know objects definitely won't hit until 3/4 of the way through the timestep, no need to multisample all the way through the first 3/4 of the timestep, might as well jump right there).